optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/factory/MoveIteratorFactoryToMoveSelectorBridge.java

```diff
@@ -51,12 +51,14 @@ public boolean supportsPhaseAndSolverCaching() {
 
     @Override
     public void phaseStarted(AbstractPhaseScope<Solution_> phaseScope) {
-        super.phaseStarted(phaseScope);
         scoreDirector = phaseScope.getScoreDirector();
+        super.phaseStarted(phaseScope);
```

Comment:
What is the reason for changing the order of these operations?

Reply:
It's like pealing an union and then putting it back together: outside in, inside out.
if you do start A, start B, start C with B depending on A, then it's better to do end C, end B, end A, then the other way around